pkg/ipamd/rpc_handler.go

```diff
@@ -198,16 +198,11 @@ func (s *server) DelNetwork(ctx context.Context, in *rpc.DelNetworkRequest) (*rp
 	eni, ip, deviceNumber, mismatchedStore, err := s.ipamContext.dataStore.UnassignPodIPv4Address(ipamKey)
 	if eni != nil && mismatchedStore {
 		if s.ipamContext.enableIpv4PrefixDelegation {
-			addr := eni.IPv4Addresses[ip]
-			if addr != nil && addr.Prefix == nil {
+			cidr := net.IPNet{IP: net.ParseIP(ip), Mask: net.IPv4Mask(255, 255, 255, 255)}
+			addr := eni.AvailableIPv4Cidrs[cidr.String()].IPv4Addresses[ip]
+			if addr != nil && addr.Prefix.IP == nil {
```

Comment:
We might not need these checks here as the above check guarantees Prefix.IP is nil for this case